pkg/operator/configobservation/network/observe_network.go

```diff
@@ -101,3 +101,77 @@ func ObserveServicesSubnet(genericListers configobserver.Listers, recorder event
 
 	return out, errs
 }
+
+func ObserveExternalIPs(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+
+	// set observed values
+	//  admission:
+	//    pluginConfig:
+	//      network.openshift.io/ExternalIPRanger:
+	//        configuration:
+	//          version: network.openshift.io/v1
+	//          kind: ExternalIPRangerAdmissionConfig
+	//          externalIPNetworkCIDRs: [...]
+	configPath := []string{"admission", "pluginConfig", "network.openshift.io/ExternalIPRanger", "configuration"}
+
+	// Need to handle 3 cases:
+	// 1. retrieval error: pass through existing, if it exists
+	// 2. Null egress policy: disable the admission controller by returning no config
+	// 3. Egress policy: enable the egress controller, pass through configuration.
+
+	errs := []error{}
+	externalIPPolicy, err := network.GetExternalIPPolicy(listers.NetworkLister, recorder)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	ingressIPs, err := network.GetIngressIPs(listers.NetworkLister, recorder)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	// Case 1: retrieval error. Pass through our existing configuration path,
+	// if it exists.
+	if len(errs) > 0 {
+		previouslyObservedConfig := map[string]interface{}{}
+
+		// an error here is very unlikely
+		prevConfig, ok, err2 := unstructured.NestedMap(existingConfig, configPath...)
+		if ok && err2 != nil {
+			unstructured.SetNestedMap(previouslyObservedConfig, prevConfig, configPath...)
+		}
+
+		return previouslyObservedConfig, errs
+	}
+
+	// Case 2: no policy: return nil, go away
+	observedConfig := map[string]interface{}{}
+	if externalIPPolicy == nil {
+		return observedConfig, nil
+	}
+
+	// Case 3: Policy, synthesize config
+	// Simply by creating this configuration, the admission controller will
+	// be enabled and block all IPs by default.
+	admissionControllerConfig := unstructured.Unstructured{}
+	admissionControllerConfig.SetAPIVersion("network.openshift.io/v1")
+	admissionControllerConfig.SetKind("ExternalIPRangerAdmissionConfig")
+
+	conf := []string{}
+	for _, cidr := range externalIPPolicy.RejectedCIDRs {
+		conf = append(conf, "!"+cidr)
+	}
+	for _, cidr := range externalIPPolicy.AllowedCIDRs {
+		conf = append(conf, cidr)
+	}
+
+	allowIngressIP := len(ingressIPs) > 0
```

Comment:
Is this how we set the option in 3.x?

Reply:
more-or-less: https://github.com/openshift/origin/blob/release-3.11/pkg/cmd/openshift-kube-apiserver/conversion.go#L30